Foundation/Data.swift

```diff
@@ -12,7 +12,7 @@
 
 #if DEPLOYMENT_RUNTIME_SWIFT
 
-#if os(Linux)
+#if os(Linux) || os(Windows)
```

Comment:
I think my comment from https://github.com/apple/swift-corelibs-foundation/pull/1868#discussion_r252576121 is still valid and this should be `#if !canImport(Darwin)`.

Reply:
mmm, pretty sure that FreeBSD also provides `malloc_good_size` ... so, I can spell it out as the inverse if you like, but I think that this is better.

Reply:
I agree that we should be more general where possible. `!canImport(Darwin) && !os(FreeBSD)`

Reply:
According to the man pages at https://man.openbsd.org/FreeBSD-12.0/malloc_good_size (which seem to support multiple OSes) no other OS incl FreeBSD has `malloc_good_size` even though it says FreeBSD's `malloc` page shows it uses jemalloc.

Reply:
Is there support for `os(FreeBSD)` ?

Reply:
Yes, there is support for `os(FreeBSD)`

Reply:
We talked a bit off PR.

I think that this block relies on (not having) the interface of the API offered by the `Darwin` module. So since the Darwin module is imported earlier if `canImport(Darwin)` is true, and we are assuming that the Darwin API is implicitly coming from that module, then `!canImport(Darwin)` is correct here.

We also looked into FreeBSD docs, and it looks like it doesn't expose exactly that symbol. (`jemalloc`, its malloc implementation, may have a similar one, but it isn't documented API). Let's just use `!canImport(Darwin)` here, and let FreeBSD port champions see if a different implementation can be used.